sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousReceiveWork.java

```diff
@@ -96,8 +104,12 @@ boolean isTerminal() {
      */
     void next(ServiceBusReceivedMessageContext message) {
         try {
+            if (timeoutBeforeNextMessageOperation != null && !timeoutBeforeNextMessageOperation.isDisposed()) {
```

Comment:
This isn't reactive to have to create a new Mono operation everytime a message come in. How about a combination of `Flux.swithOnNext(publisher).takeUntil(notCancelled)` where a new item is emitted from the `publisher` if there is a next message? And for each item emitted, delay at an interval. So if the item is emitted, then cancel the emitter.